### PR TITLE
Fix hvx_shared_object target feature flag

### PR DESCRIPTION
**Pull request by dsharletg**
This change updates the Halide runtime to use Hexagon tools 8.0.10, and fixes the hvx_shared_object target flag (currently broken because there was a breaking change in LLVM IR between LLVM trunk and hexagon-clang). The fix is to compile all the way to an object, and pass that to hexagon-clang instead of LLVM IR.

### CODE REVIEW

**Comment by ronlieb** [comment]
lgtm

File: src/HexagonOffload.cpp [hunk 1]
```diff
@@ -263,18 +263,29 @@ class InjectHexagonRpc : public IRMutator {
...
+            llvm::raw_svector_ostream assembly_stream(assembly);
+            compile_llvm_module_to_assembly(*llvm_module, assembly_stream);
+            debug(2) << assembly.c_str() << "\n";
+        }
+
         // Determine relocation mode. if both are false, its object relocation.
         bool use_shared_object = device_code.target().has_feature(Target::HVX_shared_object);
 
         if (use_shared_object) {
             // Dump the llvm module to a temp file as .ll
```
Comment (steven-johnson) [inline src/HexagonOffload.cpp]: Comment is now wrong

File: src/HexagonOffload.cpp [hunk 2]
```diff
@@ -292,7 +303,7 @@ class InjectHexagonRpc : public IRMutator {
             }
 
             hex_command += " ";
-            hex_command += tmp_bitcode.pathname();
+            hex_command += tmp_object.pathname();
             if (0) { // This path should also work, if we want to use PIC code
```
Comment (steven-johnson) [inline src/HexagonOffload.cpp]: Nit: better to comment out dead-code-left-as-example than to use if(0)

**Comment by zvookin** [comment]
Gonna merge so I can merge in to submodule_lowering2 and get a PR opened. I think one of Steven's comments is already addressed there and I'll kill the "if (0)" thing on that merge.

**Pull request merged**